exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,305 @@
+      handle_signal(signal);
+    }
+  }
+
+  auto get_signals() -> void {
+    ipc_client_.signals([&](const std::vector<tfc::ipc_ruler::signal>& signals) {
+      for (const tfc::ipc_ruler::signal& signal : signals) {
+        active_signals_.push_back(signal);
+      }
+    });
+
+    ctx_.run_for(std::chrono::milliseconds(20));
+  }
+
+  // This function removes signals that are banned
+  auto clean_signals() -> void {
+    active_signals_.erase(std::remove_if(active_signals_.begin(), active_signals_.end(),
+                                         [&](const tfc::ipc_ruler::signal& signal) {
+                                           return std::ranges::any_of(banned_signals_.begin(), banned_signals_.end(),
+                                                                      [&](const std::string& banned_string) {
+                                                                        return signal.name.find(banned_string) !=
+                                                                               std::string::npos;
+                                                                      });
+                                         }),
+                          active_signals_.end());
+  }
+
+  // This function checks the type of the signal in order to determine how to read from the slot
+  auto handle_signal(tfc::ipc_ruler::signal& signal) -> void {
+    switch (signal.type) {
+      case details::type_e::_bool: {
+        run_slot<tfc::ipc::details::type_bool, bool>(signal);
+        break;
+      }
+      case details::type_e::_int64_t: {
+        run_slot<tfc::ipc::details::type_int, int>(signal);
+        break;
+      }
+      case details::type_e::_uint64_t: {
+        run_slot<tfc::ipc::details::type_uint, uint>(signal);
+        break;
+      }
+      case details::type_e::_double_t: {
+        run_slot<tfc::ipc::details::type_double, double>(signal);
+        break;
+      }
+      case details::type_e::_string: {
+        run_slot<tfc::ipc::details::type_string, std::string>(signal);
+        break;
+      }
+      case details::type_e::_json: {
+        run_slot<tfc::ipc::details::type_json, std::string>(signal);
+        break;
+      }
+      case details::type_e::unknown: {
+        break;
+      }
+    }
+  }
+
+  // This function runs the coroutine for the slot
+  template <typename slot_type, typename value_type>
+  auto run_slot(tfc::ipc_ruler::signal& signal) -> void {
+    slots_.push_back(std::make_shared<details::slot<slot_type>>(ctx_, signal.name));
+    auto& slot = std::get<std::shared_ptr<details::slot<slot_type>>>(slots_.back());
+    slot->connect(signal.name);
+    asio::co_spawn(mqtt_client_->strand(), slot_coroutine<slot_type, value_type>(*slot, signal.name), asio::detached);
+  }
+
+  // This function stops the coroutines that are currently running and then restarts them
+  auto restart() -> void {
+    stop_coroutine_ = true;
+    ctx_.run_for(std::chrono::milliseconds(20));
```

Comment:
why is this needed?

Reply:
If this variable is left out and the ctx isn't allowed to run then the old slots are not cancelled.

Reply:
I mean the run_for, is it the same as the other one? blocking the consumer of the restart function?

Reply:
this has changed, only new signals are added.